src/main/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoImpl.java

```diff
@@ -174,6 +180,35 @@ public void deleteMessage(String parentGroup, String forumName, String creator,
         documentAdapter.commitBatch(batch);
     }
 
+    @Override
+    public void reportMessage(String parentGroup, String forumName, String creator, String reporter, String date)
+        throws DatabaseAccessException, DocumentException, InvalidOperationException {
+        if (creator.equals(reporter)) {
+            throw new InvalidOperationException("409", "A message creator can't report its message");
+        }
+        DocumentSnapshot messageSnapshot = getForumMessage(parentGroup, forumName, creator, date);
+        WriteBatch batch = documentAdapter.batch();
+        batch.update(messageSnapshot.getReference(), REPORTED_BY_FIELD, FieldValue.arrayUnion(reporter));
+        ArrayList<String> messages = (ArrayList<String>) messageSnapshot.get(REPORTED_BY_FIELD);
+        if (messages != null && messages.contains(reporter)) {
+            throw new InvalidOperationException("409", "This user already reported the message");
+        }
+        if (messages != null && messages.size() > COUNTER-1) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['-' is not followed by whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5633182)